### PR TITLE
Migrate playground-plugin to Kotlin

### PR DESCRIPTION
**Pull request by liutikas**
## Proposed Changes

Migrate playground-plugin to Kotlin

## Testing

Test: ./gradlew tasks in activity/

### CODE REVIEW

File: playground-common/playground-plugin/src/main/kotlin/androidx/playground/GradleEnterpriseConventionsPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                capture.isTaskInputFiles = true
+
+                GradleWorkaround.obfuscate(obfuscation)
+            }
+        }
+
+        settings.buildCache.remote(HttpBuildCache::class.java) { remote ->
+            remote.url = URI("https://ge.androidx.dev/cache/")
+            val buildCachePassword = System.getenv("GRADLE_BUILD_CACHE_PASSWORD")
+            if (isCI && buildCachePassword != null && buildCachePassword.isNotEmpty()) {
```
Comment (dlam) [inline playground-common/playground-plugin/src/main/kotlin/androidx/playground/GradleEnterpriseConventionsPlugin.kt]: Can simplify to `!String.isNullOrEmpty()`

```suggestion
            if (isCI && !buildCachePassword.isNullOrEmpty()) {
```

File: playground-common/playground-plugin/src/main/kotlin/androidx/playground/PlaygroundPlugin.kt [hunk 2]
```diff
@@ -13,38 +13,39 @@
...
         validateJvm(settings)
     }
 
-    def validateJvm(Settings settings) {
+    private fun validateJvm(settings: Settings) {
         // validate JVM version to print an understandable error if it is not set to the
         // required value (11)
-        def jvmVersion = System.getProperty("java.vm.specification.version")
+        val jvmVersion = System.getProperty("java.vm.specification.version")
         if (jvmVersion != "11") {
```
Comment (dlam) [inline playground-common/playground-plugin/src/main/kotlin/androidx/playground/PlaygroundPlugin.kt]: Could just make this a check with a string template:

```
check(jvmVersion == "11") {
    """
    AndroidX build must be invoked with JDK 11.
    ${
        if (settings.gradle.startParameter.projectProperties.containsKey("android.injected.invoked.from.ide")) {
            "Make sure to set the gradle JDK to JDK 11 in the project settings. (File -> Other Settings -> Default Project Structure)"
        } else {
            "Make sure your JAVA_HOME environment variable points to Java 11 JDK."
        }
    }
    Current version: $jvmVersion
    Current JAVA HOME: ${System.getProperty("java.home")}
    """.trimIndent()
}
```

At the very least I think we should add `.trimIndent()` / `.trimMargin()` to cut down the extra whitespace.

File: playground-common/playground-plugin/src/main/kotlin/androidx/playground/GradleEnterpriseConventionsPlugin.kt (hunk 1 above)
Comment (liutikas) [reply playground-common/playground-plugin/src/main/kotlin/androidx/playground/GradleEnterpriseConventionsPlugin.kt]: Done

File: playground-common/playground-plugin/src/main/kotlin/androidx/playground/PlaygroundPlugin.kt (hunk 2 above)
Comment (liutikas) [reply playground-common/playground-plugin/src/main/kotlin/androidx/playground/PlaygroundPlugin.kt]: Done

**Comment by liutikas** [comment]
Turns out this breaks things. I'll look at it further tomorrow.

**Review by dlam — APPROVED** [review]

**Pull request closed without merging**